lib/features/rectangles/data/datasources/rectangle_image_remote_data_source.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:the_boxesv3/core/error/exceptions.dart';
+import 'package:the_boxesv3/features/rectangles/data/models/rectangle_image_model.dart';
+import 'package:http/http.dart' as http;
+
+abstract class RectangleImageRemoteDataSource {
+  /// calls the https://picsum.photos/v2/list?limit={count} endpoint
+  ///
+  /// Throws a [ServerException] for all error codes
+  Future<List<RectangleImageModel>> getRectangleImageList(int count);
+  Future<RectangleImageModel> getRandomRectangleImage();
+}
+
+class RectangleImageRemoteDataSourceImpl
+    implements RectangleImageRemoteDataSource {
+  final http.Client client;
+
+  RectangleImageRemoteDataSourceImpl({required this.client});
+
+  @override
+  Future<List<RectangleImageModel>> getRectangleImageList(int count) async {
+    var url =
+        Uri.https('picsum.photos', 'v2/list', {'limit': count.toString()});
+    final response = await client.get(url);
+    if (response.statusCode == 200) {
+      List<RectangleImageModel> imageList = [];
+      List body = jsonDecode(response.body);
+      for (var i = 0; i < body.length; i++) {

```

Comment:
find a way to do it in model

Reply:
static function which would return a list of images

Reply:
will do